GitHubReactions.ps1

```diff
@@ -0,0 +1,363 @@
+
+    .PARAMETER Uri
+        Uri for the repository.
+        The OwnerName and RepositoryName will be extracted from here instead of needing to provide
+        them individually.
+
+    .PARAMETER Issue
+        The issue number.
+
+    .PARAMETER ReactionType
+        The type of reaction you want to retrieve. This is aslo called the 'content' in the GitHub API.
+        Valid options are based off: https://developer.github.com/v3/reactions/#reaction-types
+
+    .PARAMETER AccessToken
+        If provided, this will be used as the AccessToken for authentication with the
+        REST Api.  Otherwise, will attempt to use the configured value or will run unauthenticated.
+
+    .PARAMETER NoStatus
+        If this switch is specified, long-running commands will run on the main thread
+        with no commandline status update.  When not specified, those commands run in
+        the background, enabling the command prompt to provide status information.
+        If not supplied here, the DefaultNoStatus configuration property value will be used.
+
+    .EXAMPLE
+        Get-GitHubReaction -OwnerName Microsoft -RepositoryName PowerShellForGitHub -Issue 157
+
+        Gets the reactions for issue 157 from the Microsoft\PowerShellForGitHub project.
+
+    .EXAMPLE
+        Get-GitHubReaction -OwnerName Microsoft -RepositoryName PowerShellForGitHub -Issue 157 -ReactionType eyes
+
+        Gets the 'eyes' reactions for issue 157 from the Microsoft\PowerShellForGitHub project.
+
+    .EXAMPLE
+        Get-GitHubIssue -OwnerName Microsoft -RepositoryName PowerShellForGitHub -Issue 157 | Get-GitHubReaction
+
+        Gets a GitHub issue and pipe it into Get-GitHubReaction to get all the reactions for that issue.
+
+    .NOTES
+        Currently, this only supports reacting to issues. Issue comments, commit comments and PR comments will come later.
+
+    .NOTES
+        The alias parameters 'number' and 'repository_url' are so that this cmdlet composes with `Get-GitHubIssue`.
+#>
+    [CmdletBinding(
+        SupportsShouldProcess,
+        DefaultParameterSetName='Elements')]
+    [Diagnostics.CodeAnalysis.SuppressMessageAttribute("PSShouldProcess", "", Justification="Methods called within here make use of PSShouldProcess, and the switch is passed on to them inherently.")]
+    param(
+        [Parameter(Mandatory, ParameterSetName='Elements')]
+        [string] $OwnerName,
+
+        [Parameter(Mandatory, ParameterSetName='Elements')]
+        [string] $RepositoryName,
+
+        [Parameter(Mandatory, ParameterSetName='Uri', ValueFromPipelineByPropertyName)]
+        [Alias("repository_url")]
+        [string] $Uri,
+
+        [Parameter(Mandatory, ParameterSetName='Elements')]
+        [Parameter(Mandatory, ParameterSetName='Uri', ValueFromPipelineByPropertyName)]
+        [Alias("number")]
+        [int64] $Issue,
+
+        [ValidateSet('+1', '-1', 'laugh', 'confused', 'heart', 'hooray', 'rocket', 'eyes')]
```

Comment:
PowerShell standard for parameters is SentenceCase.  You'll need to `.ToLower()` the value later.

```suggestion
        [ValidateSet('+1', '-1', 'Laugh', 'Confused', 'Heart', 'Hooray', 'Rocket', 'Eyes')]
```